core/modules/czar/Czar.cc

```diff
@@ -134,6 +136,9 @@ Czar::submitQuery(std::string const& query,
     std::string const msgTableName = "message_" + userQueryId;
     std::string const lockName = resultDb + "." + msgTableName;
 
+    // Add logging context with user query ID
+    LOG_MDC_SCOPE("TID", userQueryId);
```

Comment:
Curious: How is TID different from QID?

Reply:
We just had 1-hour discussion yesterday about how we want to get rid of TID 🙂 TID is a sort of temporary ID which is assigned to a request before the request is registered in QMeta and gets it actual QID. Some requests never go to QMeta, so for those we only have TID. The whole things is a mess and we'd like to get rid of TID, but it's not quite trivial.

Reply:
Just to add - TID is a sort of randomly generated but not unique and not persistent, QID is unique and persistent.